guessit/api.py

```diff
@@ -154,7 +169,7 @@ def configure(self, options=None, rules_builder=rebulk_builder, force=False, san
                               self.advanced_config != advanced_config
 
         if should_build_rebulk:
-            self.advanced_config = advanced_config
+            self.advanced_config = deepcopy(advanced_config)
```

Comment:
Key change to fix the issue is here. other changes are tests and refactoring to help testing.